ts/Series/SeriesOnPointComposition.ts

```diff
@@ -0,0 +1,517 @@
+        (drawPointsFunctions as any)[this.type + 'DrawPoints'].call(this);
+
+        this.seriesDrawConnector();
+    }
+
+    /* *
+     *
+     *  Classes
+     *
+     * */
+
+    /**
+     * @private
+     */
+    export class Additions {
+
+        /* *
+         *
+         *  Constructors
+         *
+         * */
+
+        /**
+         * @private
+         */
+        public constructor(series: SeriesComposition) {
+            this.series = series;
+            this.options = series.options.onPoint;
+        }
+
+        /* *
+         *
+         *  Properties
+         *
+         * */
+
+        public series: SeriesComposition;
+
+        public connector?: SVGElement = void 0 as any;
+
+        public options?: OnPoint;
+    }
+
+}
+
+/* *
+ *
+ *  Default Export
+ *
+ * */
+
+export default SeriesOnPointComposition;
+
+/* *
+ *
+ *  API Options
+ *
+ * */
+
+/**
+ * Color of a connector line. By default it's the series' color.
+ *
+ * @requires    modules/series-on-point
+ * @since       next
+ * @type        {string}
+ * @apioption   plotOptions.series.onPoint.connectorOptions.color
+ */
+
+/**
+ * A name for the dash style to use for connector.
+ *
+ * @requires    modules/series-on-point
+ * @since       next
+ * @type        string
+ * @apioption   plotOptions.series.onPoint.connectorOptions.dashStyle
+ */
+
+/**
+ * Pixel width of a connector line.
+ *
+ * @default     1
+ * @requires    modules/series-on-point
+ * @type        {number}
+ * @since       next
+ * @product     highcharts highstock
+ * @apioption   plotOptions.series.onPoint.connectorOptions.width
+*/
+
+/**
+ * An id of the point that we connect series to.
+
+ * @requires   modules/series-on-point
+ * @since      next
+ * @type       {string}
+ * @apioption  plotOptions.series.onPoint.id
+ */
+
+/**
+ * Series center offset from the original x position. If defined, the connector
+ * line is drawn connecting original position with new position.
+ *
+ * @requires   modules/series-on-point
+ * @since      next
+ * @type       {number}
+ * @apioption  plotOptions.series.onPoint.position.offsetX
+ */
+
+/**
+ * Series center offset from the original y position. If defined, the connector
+ * line is drawn from original position to a new position.
+ *
+ * @requires   modules/series-on-point
+ * @since      next
+ * @type       {number}
+ * @apioption  plotOptions.series.onPoint.position.offsetY
+ */
+
+/**
+ * X position of the series center. By default, the series is displayed on a
+ * point that it's connected to.
+ *
+ * @requires   modules/series-on-point
+ * @since      next
+ * @type       {number}
+ * @apioption  plotOptions.series.onPoint.position.x
+ */
+
+/**
+ * Y position of the series center. By default, the series is displayed on a
+ * point that it's connected to.
+ *
+ * @requires   modules/series-on-point
+ * @since      next
+ * @type       {number}
+ * @apioption  plotOptions.series.onPoint.position.y
+ */
+
```

Comment:
It would be nice to have some API docs for onPoint - the main object. See marker on any series type for comparison. Same for `onPoint.position` and `onPoint.connectorOptions`.